### ISSUE TITLE
Get method Form appending searchParams to URL instead of updating them

### ISSUE THREAD

**Issue by lswest** [remix-run/remix#4112]
### What version of Remix are you using?

1.7.0

### Steps to Reproduce

1. Create a remix project
2. Add the following to a page under app/routes:
```jsx
import { Form } from "@remix-run/react";

export default function ExamplePage() {
  return (
    <Form>
      <select name="filter">
        <option value="">Select a filter</option>
        <option value="all">All</option>
        <option value="active">Active</option>
      </select>
      <button type="submit">Filter</button>
    </Form>
  )
}
```
3. Start dev server
4. Navigate to the page under http://localhost:3000/ (I did this in Chrome, I didn't test other browsers)
5. Select a filter (for example, "All") and hit the submit button.
6. Select another filter (for example, "Active") and hit the submit button.
7. [OPTIONAL] Repeat selections

See [reproduction](https://github.com/lswest/remix-get-form-repro)

### Expected Behavior

The URL should contain `?filter=active`.

### Actual Behavior

The URL contains `?filter=all&filter=active` after first reselection.

The URL contains `?filter=all&filter=active&filter=all` after another selection, etc.

**Comment by lswest** [comment]
Took a bit of a look through the node_modules and played around with some tweaks. The issue is caused by line 1226 in components.tsx in remix-react (see [here](https://github.com/remix-run/remix/blob/main/packages/remix-react/components.tsx#L1226)).

Looks like it was changed in December in this commit: f1c7c874a1a71835e5aab54e9c0fbf6093daf6fb

Basically, the old value is in the url.searchParams before the for loop, and then we append a new instance with the new value (which is fine if that's what you want and are using .getAll() when parsing the searchParams) when we submit the form.

Presumably this change was made to avoid overwriting values with the same key - I'm just not sure if this particular use-case was intended to keep working or not. I'm also open to suggestions on how to get the old behavior back. Maybe write a custom submit?

**Comment by Josh865** [comment]
I'm having the same issue running the app locally, but I haven't been able to reproduce it on Stackblitz. For example, [this works fine](https://stackblitz.com/edit/node-4w3hzz?file=app%2Froutes%2Findex.tsx), but the same code (with only some Tailwind classes added) produces the following locally:

https://user-images.githubusercontent.com/31568312/187992722-d90d0289-d33d-4dd2-af21-9252a862e4fe.mov

**Comment by lswest** [comment]
@Josh865 The problem occurs for us in production too at our company (in Docker, on our own server, using Remix v1.7.0). Is Stackblitz using an earlier version of Remix, perhaps?

**Comment by lswest** [comment]
Seems to be solved/known already. See this PR: #4046 

The workaround/fix seems to be to set action=".", which avoids passing in existing searchParams. See [this comment](https://github.com/remix-run/remix/pull/4046#issuecomment-1227500836)

Anyone from the Remix team know if this will appear in the docs (or already does)? I couldn't find a page referencing this, though I did see a short reference to this in a changelog, but not in a context where I was able to put 2 and 2 together.

I'm leaving this item open for the moment in the hopes of getting some information about docs/if this is a permanent change or not, but it is technically solved with the above change.
